src/lua/httpc.lua

```diff
@@ -32,8 +32,18 @@
 local driver = package.loaded.http.client
```

Comment:
Regarding the commit message (1st patch):

> Header with content type will be set to
> "application/json" if it was not defined by user in request's options.

If I wouldn't know our agreements I would read it as if the implementation doing so always if the `Content-Type` header is not set. I think that we should mention here that it is only when the body is encoded as JSON (I guess always except when a string is passed -- please, correct me if I'm wrong here).

BTW, I looked at the implementation and I don't see, where we set the header. Am I missed anything?

Reply:
> <...> only when the body is encoded as JSON (I guess always except when a string is passed -- please, correct me if I'm wrong here).

I actually was wrong. A table, a userdata, a cdata is encoded as JSON and so the header is added (if not set) only when a value of one of those types is passed as a body.

The rest of the comment is actual:

* It is unclear, when exactly the header is set.
* I don't see an implementation of this logic.

Reply:
I like how the description of the encoding functionality looks now. Thanks!